dandi/dandiapi.py

```diff
@@ -436,19 +436,21 @@ def publish(self) -> "RemoteDandiset":
         `Version`.
         """
         lgr.debug("Waiting for Dandiset %s to complete validation ...", self.identifier)
-        while True:
+        start = time()
+        # Wait 20s for celery tasks to finish
+        while time() - start < 20:
             r = self.client.get(f"{self.version_api_path}info/")
             if "status" not in r:
                 # Running against older version of dandi-api that doesn't
                 # validate
                 break
             if r["status"] == "Valid":
                 break
-            elif r["status"] == "Invalid":
```

Comment:
While a time limit is useful here, I think that this check should be kept so that Dandisets deemed invalid are reported as such quickly.  The only reason this check is a problem is because dandi-api is setting the status to "Invalid" without a helpful `validation_error`.

Reply:
@jwodder why do you say that it is only the absence of validation_error which is the issue?
```
_________________________ test_publish_and_manipulate __________________________
/home/runner/work/dandi-cli/dandi-cli/dandi/tests/test_dandiapi.py:69: in test_publish_and_manipulate
    version_id = d.publish().version.identifier
/home/runner/work/dandi-cli/dandi-cli/dandi/dandiapi.py:551: in publish
    f"Dandiset {self.identifier} is invalid: {r['validation_error']}"
E   ValueError: Dandiset 000002 is invalid:
```
so it does `format` the string ok, and it seems a matter of receiving `"Invalid"` status whenever it should not be "Invalid" yet.  

There must be some explicit way API would say that status is "Pending" or "Unknown".  It must not return "Invalid" if it just "has not decided yet", which is what I guess happens here.

Reply:
There is a "Pending" status, but @dchiquito is saying that there's no decent way to set the status back to that during the validation lifecycle.